src/command/projections/devicecount-legal-entity.projection.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { AbstractProcessor } from '../../commons/event-processing/abstract.processor';
+import { Event } from '../../commons/event-store/event';
+import { LegalEntityRemoved } from '../../commons/events/legal-entity/removed';
+import { DeviceRegistered } from '../../commons/events/sensordevice/device/registered';
+import { DeviceRemoved } from '../../commons/events/sensordevice/device/removed';
+import { DeviceCountDeviceEsListener } from './devicecount-device.listener';
+import { DeviceCountLegalEntityEsListener } from './devicecount-legal-entity.listener';
+import { ILegalEntityDeviceCount } from './models/legalentity-device-count.schema';
+
+@Injectable()
+export class DeviceCountLegalEntityProjection extends AbstractProcessor {
+    constructor(
+        protected readonly deviceListener: DeviceCountDeviceEsListener,
+        protected readonly legalEntityListener: DeviceCountLegalEntityEsListener,
+        @InjectModel('LegalEntityDeviceCount') private legalEntityDeviceCountModel: Model<ILegalEntityDeviceCount>,
+    ) {
+        super([deviceListener, legalEntityListener]);
+    }
+
+    async process(event: Event, originSync: boolean): Promise<void> {
+        if (!originSync) {
+            if (event instanceof DeviceRegistered) {
```

Comment:
Hier zit wel de nodige code duplicatie in ... Dit moet toch (gemakkelijk) mooier kunnen? :wink:

Reply:
De filter kan ik er wel buiten halen, maar die is niet altijd nodig (als events niet van het type DeviceRegistered, DeviceRemoved / LegalEntityRemoved zijn).